src/Interpreters/Access/InterpreterCreateUserQuery.cpp

```diff
@@ -100,9 +100,14 @@ BlockIO InterpreterCreateUserQuery::execute()
     auto & access_control = getContext()->getAccessControl();
     auto access = getContext()->getAccess();
     access->checkAccess(query.alter ? AccessType::ALTER_USER : AccessType::CREATE_USER);
+    bool implicit_no_password_allowed = access_control.isImplicitNoPasswordAllowed();
     bool no_password_allowed = access_control.isNoPasswordAllowed();
     bool plaintext_password_allowed = access_control.isPlaintextPasswordAllowed();
 
+     if (!query.auth_data && !implicit_no_password_allowed)
```

Comment:
`InterpreterCreateUserQuery` is used not only for `CREATE USER` query but also for `ALTER USER` and even `ATTACH USER`. It's not correct to put this restriction on `ALTER QUERY` (otherwise we won't be able to alter settings without altering authentication data). And probably it's better not to put restriction on `ATTACH USER` either.